internal/controller/policymanifest_controller.go

```diff
@@ -0,0 +1,133 @@
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	giantswarmPolicy "github.com/giantswarm/kyverno-policy-operator/api/v1alpha1"
+)
+
+// PolicyManifestReconciler reconciles a PolicyManifest object
+type PolicyManifestReconciler struct {
+	client.Client
+	Scheme               *runtime.Scheme
+	Log                  logr.Logger
+	DestinationNamespace string
+	Background           bool
+}
+
+//+kubebuilder:rbac:groups=giantswarm.io,resources=policymanifests,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=giantswarm.io,resources=policymanifests/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=giantswarm.io,resources=policymanifests/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the PolicyManifest object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.17.0/pkg/reconcile
+func (r *PolicyManifestReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	var polman exceptionRecommender.PolicyManifest
+
+	if err := r.Get(ctx, req.NamespacedName, &polman); err != nil {
+		// Error fetching the policy manifest
+
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+
+		log.Log.Error(err, "unable to fetch policy manifest")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// TODO(user): your logic here
+
+	// kpe stands for Kyverno Policy Exception
+	kpe := kyvernov2beta1.PolicyException{}
+	// Set kpe destination namespace.
+	kpe.Namespace = r.DestinationNamespace
+	// Set kpe name.
+	kpe.Name = fmt.Sprintf("gs-kpo-%s-exceptions", polman.ObjectMeta.Name)
+	// Set labels.
+	kpe.Labels = make(map[string]string)
+	kpe.Labels["app.kubernetes.io/managed-by"] = "kyverno-policy-operator"
+	kpe.Labels["policy.giantswarm.io/policy"] = polman.ObjectMeta.Labels["policy.giantswarm.io/policy"]
+
+	kpe.Spec.Background = &r.Background
+
+	allTargets := make([]giantswarmPolicy.Target, len(polman.Spec.Exceptions)+len(polman.Spec.AutomatedExceptions))
+	copy(allTargets, polman.Spec.Exceptions)
+	copy(allTargets[len(polman.Spec.Exceptions):], polman.Spec.AutomatedExceptions)
+
+	policyMap := make(map[string]kyvernov1.ClusterPolicy)
+	var kyvernoPolicy kyvernov1.ClusterPolicy
+	if err := r.Get(ctx, types.NamespacedName{Namespace: "", Name: polman.Name}, &kyvernoPolicy); err != nil {
+		// Error fetching the report
+		log.Log.Error(err, "unable to fetch Kyverno Policy")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	policyMap[polman.Name] = kyvernoPolicy
```

Comment:
Since we reconcile ClusterPolicies, using a cache would be better than going to the api server every time